experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,299 @@
+end
+
+function is_hyperelliptic(o::Origami)
+    # check whether -1 is in the veech group
+	if !veech_group_is_even(o) 
+        return false
+    end
+
+    n = 2
+    n_inv = 1 / 2
+	x = horizontal_perm(o)
+	y = vertical_perm(o)
+	g = genus(o)
+
+    L = point_reflections(o)
+	L = filter(i -> order(i) == 2, L)
+
+	if L == []
+        return false
+    end
+
+    # helper to find all elements in x that are not in y
+    function list_diff(x, y)
+        set_1 = Set(x)
+        set_2 = Set(y)
+        diff = setdiff(set_1, set_2)
+        return collect(diff)
+    end
+
+    degree_list = collect(1:degree(o))
+    for sigma in L
+        # fixpoints
+        b = 0
+        b = b + length(list_diff(degree_list, moved_points(sigma)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*x)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*y)))
+
+        for i in degree_list
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
+            return true
+        end
+    end
+
+    return false
+end
+
+function cylinder_structure(o::Origami)
+    # TODO returns Vector{Any}, maybe cast to Integer?
+    gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+    return GAP.gap_to_julia(gap_obj)
+end
+
+function veech_group_and_orbit(o::Origami)
+    return GAP.Globals.VeechGroupAndOrbit(GapObj(o))
+end
+
+function veech_group_is_even(o::Origami)
+    return GAP.Globals.VeechGroupIsEven(GapObj(o))
+end
+
+function are_equivalent(o1::Origami, o2::Origami)
+    return GAP.Globals.OrigamisEquivalent(GapObj(o1), GapObj(o2))
+end
+
+# TODO why is this not in canonical.jl?
+function normalform_conjugators(o::Origami)
+    x = horizontal_perm(o)
+    y = vertical_perm(o)
+    n = degree(o)
+    G = []
+
+    # Starting from each of the vertices found above, do a breadth-first search
+	# and list the vertices in the order they appear.
+	# This defines a permutation l with which we conjugate x and y.
+	# From the resulting list of pairs of permutations (all of which are by
+	# definition simultaneously conjugated to (x,y)) we choose the
+	# lexicographically smallest one as the canonical form.
+    for i in 1:n
+        L = fill(0, n)
+        seen = fill(false, n)
+        Q = [i]
+        seen[i] = true
+        numSeen = 1
+        L[i] = 1
+        while numSeen < n
+            v = popfirst!(Q)
+            wx = v^x
+            wy = v^y
+            if !seen[wx]
+                push!(Q, wx)
+                seen[wx] = true
+                numSeen = numSeen + 1
+                L[wx] = numSeen
+            end
+            if !seen[wy]
+                push!(Q, wy)
+                seen[wy] = true
+                numSeen = numSeen + 1
+                L[wy] = numSeen
+            end
+        end
+        push!(G, L)
+    end
+
+    return perm.(G)
```

Comment:
Just create the permutations right away, that halves the peak memory usage, I think?
```suggestion
        push!(G, perm(L))
    end

    return G
```